### ISSUE TITLE
Missing Push Notification Entitlement

### ISSUE THREAD

**Issue by zpapp** [AFNetworking/AFNetworking#2938]
Hi,

I recently included AFNetworking in an app and I received the following warning from Apple when I submitted the app for review:

> Missing Push Notification Entitlement - Your app appears to include API used to register with the Apple Push Notification service, but the app signature's entitlements do not include the "aps-environment" entitlement. If your app uses the Apple Push Notification service, make sure your App ID is enabled for Push Notification in the Provisioning Portal, and resubmit after signing your app with a Distribution provisioning profile that includes the "aps-environment" entitlement. See "Provisioning and Development" in the Local and Push Notification Programming Guide for more information. If your app does not use the Apple Push Notification service, no action is required. You may remove the API from future submissions to stop this warning. If you use a third-party framework, you may need to contact the developer for information on removing the API. 

This app has been submitted to Apple many times before and this is the first time I got this warning, so it seems reasonable to assume that this is being caused by AFNetworking. Is this a known issue? Is there a way to get rid of this warning without removing all of AFNetworking?

Thanks,
Zsombor

**Comment by richardgroves** [comment]
I've just started getting this warning email on an app that *doesn't* use AFNetworking and where I can't work out what change I made to cause the problem - I suspect it is a false positive in Apple's testing and will be silently fixed by them at some point.

Really annoying that there doesn't seem to be any feedback available about what API is triggering the problem.

This Apple forum thread has some indication that it might be an XCode 6.4 issue  - https://forums.developer.apple.com/message/42918#42918 - and that lots of other people are seeing it.
